dagmc_stats/DagmcFile.py

```diff
@@ -106,6 +109,52 @@ def __set_entityset_ranges(self):
                 self._my_moab_core.get_entities_by_type_and_tag(self.root_set, types.MBENTITYSET,
                                                                 self.dagmc_tags['geom_dim'], [dimension])
 
+    def get_meshset_by_id(self, dim, ids=[]):
+        """Get meshset of the geometry with specified dimension and ids
+
+        inputs
+        ------
+        dim : (Integer or String) Dimension of the meshset. 0: 'node(s)',
+                                1: 'curve(s)', 2: 'surface(s)', 3: 'volume(s)'
+        ids : (Integer) Global ID(s) of the meshset
+
+        outputs
+        -------
+        meshset : meshset of the geometry with given dimension and ids. First,
+                  dim will be checked. If dim is invalid, the root set will be
+                  returned. Then, if id is invalid (empty or not in the given
+                  dim range), all entities with the given dim will be returned.
+        """
+        if isinstance(dim, int) and dim in self.entityset_types.keys():
+            dim = self.entityset_types[dim]
+        #print(list(my_dict.keys())[list(my_dict.values()).index(112)])
+        elif type(dim) == str and dim.lower() in ['node', 'curve', 'surface', 'volume', 'nodes', 'curves', 'surfaces', 'volumes']:
+            dim = dim.lower()
+            if dim[-1] != 's':
+                dim = dim + 's'
```

Comment:
This `dim` needs to be converted to an integer before line 142

Reply:
line 142 uses the string version of `dim`, as does the rest of the function. The integer version is converted to string above this.

Reply:
oh! must have misread this